src/lobsterpy/featurize/core.py

```diff
@@ -1146,3 +1132,530 @@ def get_fingerprint_df(
+        }
+
+        for interactions in complete_data:
+            for ii, l1, l2 in zip(range(len(bin_centers)), bin_edges[:-1], bin_edges[1:]):
+                if l1 <= interactions[1] < l2:
+                    # Add icoxx values to the corresponding bin
+                    label_bwdf[bond_label]["icoxx_binned"][ii] += (
+                        interactions[3] if abs(interactions[3]) > self.interactions_tol else 0
+                    )
+                    label_bwdf[bond_label]["icoxx_counts"][ii] += (
+                        1 if abs(interactions[3]) > self.interactions_tol else 0
+                    )
+        label_bwdf["centers"] = bin_centers
+        label_bwdf["edges"] = bin_edges
+        label_bwdf["bin_width"] = self.bin_width  # type: ignore[assignment]
+
+        # Normalize BWDF data
+        return self._normalize_bwdf(bwdf=label_bwdf)
+
+    @staticmethod
+    def _get_features_col_names(bwdf: dict) -> list[str]:
+        """
+        Get the names of the features that will be generated by the class.
+
+        Args:
+            bwdf: dictionary containing the BWDF data
+
+        Returns:
+            list of feature names
+        """
+        features = []
+        for edge_1, edge_2 in zip(bwdf["edges"][:-1], bwdf["edges"][1:]):
+            features.append(f"bwdf_{round(edge_1, 2)}-{round(edge_2, 2)}")
+
+        return features
+
+    def _normalize_bwdf(self, bwdf: dict) -> dict:
+        """
+        Normalize BWDF data based on the normalization strategy.
+
+        :param bwdf: BWDF data as a dictionary
+
+        Returns:
+            Normalized BWDF data as a dictionary
+        """
+        for bwdf_label, bwdf_value in bwdf.items():
+            if bwdf_label not in ("centers", "edges", "bin_width", "wasserstein_dist_to_rdf"):
+                if self.normalization == "area":
+                    total_area = np.sum(np.abs(bwdf_value["icoxx_binned"]) * self.bin_width)
+                    bwdf[bwdf_label]["icoxx_binned"] = np.nan_to_num(bwdf_value["icoxx_binned"] / total_area)
+                elif self.normalization == "formula_units":
+                    formula_units = self.structure.composition.get_reduced_formula_and_factor()[-1]
+                    bwdf[bwdf_label]["icoxx_binned"] = np.nan_to_num(bwdf_value["icoxx_binned"] / formula_units)
+                elif self.normalization == "counts":
+                    bwdf[bwdf_label]["icoxx_binned"] = np.nan_to_num(
+                        bwdf_value["icoxx_binned"] / bwdf_value["icoxx_counts"]
+                    )
+
+        return bwdf
+
+    def get_binned_bwdf_df(self, ids: str | None = None) -> pd.DataFrame:
+        """Return a pandas dataframe with computed BWDF features as columns.
+
+        Args:
+            ids: set index name in the pandas dataframe. Default is None.
+
+        Returns:
+            A pandas dataframe object with BWDF features
```

Comment:
Again, we need more documentation here

Reply:
Added detailed descriptions now in doc-strings. Thanks for catching it